apt-hook/json-hook.cc

```diff
@@ -199,6 +199,92 @@ std::string create_count_message(security_package_counts &counts) {
     return message_ss.str();
 }
 
+enum CloudID {AWS, AZURE, GCE, NONE};
+
+CloudID get_cloud_id() {
+    std::ifstream cloud_id_file("/run/cloud-init/cloud-id");
+    CloudID ret = NONE;
+    if (cloud_id_file.is_open()) {
+        std::string cloud_id_str((std::istreambuf_iterator<char>(cloud_id_file)), (std::istreambuf_iterator<char>()));
+        if (cloud_id_str == "aws") {
+            ret = AWS;
+        } else if (cloud_id_str == "azure") {
+            ret = AZURE;
+        } else if (cloud_id_str == "gce") {
+            ret = GCE;
+        }
+        cloud_id_file.close();
+    }
+    return ret;
+}
+
+bool is_xenial() {
+    std::ifstream os_release_file("/etc/os-release");
+    bool ret = false;
+    if (os_release_file.is_open()) {
+        std::string os_release_str((std::istreambuf_iterator<char>(os_release_file)), (std::istreambuf_iterator<char>()));
+        if (os_release_str.find("xenial") != os_release_str.npos) {
```

Comment:
This method feels sort of loose and dangerous because it's merely a substring match on _any_ field. Just because a issue here would lead to a very confusing bug. But maybe making it better is not worth it.

Reply:
I would say if you have the wrong codename in /etc/os-release then there is something worse going on. And given there should not be any unusual adjective in os-release apart from the release codename, I would say we can keep this as-is.